### PR TITLE
Timeout re-implemented

### PR DESCRIPTION
**Pull request by benstoltz**
…optional time for a default

### CODE REVIEW

File: tests/dummy/app/error-handling/template.hbs [hunk 1]
```diff
@@ -29,3 +29,19 @@
...
+  type=model.type
+  datasets=invalidUrlDataset
+  series=model.series
+  timeout=timeout
+  onError=(action 'showError3')
+}}</pre>
+<div class="alert alert-danger">{{errorMessage3}}</div>
+{{cedar-chart
+  type=model.type
+  datasets=invalidUrlDataset
```
Comment (tomwayson) [inline tests/dummy/app/error-handling/template.hbs]: don't use `invalidUrlDataset` dataset for this one

File: tests/dummy/app/error-handling/template.hbs [hunk 2]
```diff
@@ -29,3 +29,19 @@
   invalidSpecMessage="invalidSpecMessage is no longer used and should show a deprecation"
   onError=(action 'showError2')
 }}
+<p>Example of a timeout</p>
+<pre>\{{cedar-chart
+  type=model.type
+  datasets=invalidUrlDataset
+  series=model.series
+  timeout=timeout
```
Comment (tomwayson) [inline tests/dummy/app/error-handling/template.hbs]: you should probably have this show that the timeout is a number, in milliseconds, i.e. `timeout=1` and then explain above that we're using one second to force it.

File: tests/dummy/app/error-handling/template.hbs [hunk 3]
```diff
@@ -29,3 +29,19 @@
   invalidSpecMessage="invalidSpecMessage is no longer used and should show a deprecation"
   onError=(action 'showError2')
 }}
+<p>Example of a timeout</p>
+<pre>\{{cedar-chart
```
Comment (tomwayson) [inline tests/dummy/app/error-handling/template.hbs]: Add a note here explaining that `timeout` is a number, in milliseconds

File: addon/components/cedar-chart/component.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
-import { scheduleOnce } from '@ember/runloop';
+import { scheduleOnce, later } from '@ember/runloop';
 import { tryInvoke } from '@ember/utils';
 import Component from '@ember/component';
 import cedar from 'cedar';
 
 export default Component.extend({
   classNames: ['cedar-chart'],
+  defaultTimeout: 7000,
```
Comment (tomwayson) [inline addon/components/cedar-chart/component.js]: I don't think we want a default timeout - if user supplies a `timeout` use that, otherwise let it ride.

**Comment by tomwayson** [comment]
For now using fetch-mock to ensure that the timeout test passes reliably.

Also added ember-exam and used it to run tests 50 times before pushing, and we can use that to flush out flakey tests in the future.

**Comment by benstoltz** [comment]
@tomwayson would there be any benefit for us to make timeoutErr be an optional input or would that be overkill?

**Pull request merged**